v2/pkg/engine/datasource/pubsub_datasource/pubsub_datasource.go

```diff
@@ -273,56 +240,52 @@ type PubSub interface {
 	// ID is the unique identifier of the pubsub implementation (e.g. NATS)
 	// This is used to uniquely identify a subscription
 	ID() string
-	// Subscribe starts listening on the given topic and sends the received messages to the given next channel
-	Subscribe(ctx context.Context, topic string, updater resolve.SubscriptionUpdater) error
-	// Publish sends the given data to the given topic
-	Publish(ctx context.Context, topic string, data []byte) error
-	// Request sends a request on the given topic and writes the response to the given writer
-	Request(ctx context.Context, topic string, data []byte, w io.Writer) error
+	// Subscribe starts listening on the given subjects and sends the received messages to the given next channel
+	Subscribe(ctx context.Context, subjects []string, updater resolve.SubscriptionUpdater) error
+	// Publish sends the given data to the given subject
+	Publish(ctx context.Context, subject string, data []byte) error
+	// Request sends a request on the given subject and writes the response to the given writer
+	Request(ctx context.Context, subject string, data []byte, w io.Writer) error
 }
 
 type SubscriptionSource struct {
 	pubSub PubSub
 }
 
 func (s *SubscriptionSource) UniqueRequestID(ctx *resolve.Context, input []byte, xxh *xxhash.Digest) error {
-	topic, err := jsonparser.GetString(input, "topic")
-	if err != nil {
-		return err
-	}
-	_, err = xxh.WriteString(topic)
-	if err != nil {
-		return err
-	}
-	_, err = xxh.WriteString(s.pubSub.ID())
+	// input must be unique across datasources
+	_, err := xxh.Write(input)
 	return err
 }
 
 func (s *SubscriptionSource) Start(ctx *resolve.Context, input []byte, updater resolve.SubscriptionUpdater) error {
-	topic, err := jsonparser.GetString(input, "topic")
+	var subjects []string
+	_, err := jsonparser.ArrayEach(input, func(value []byte, dataType jsonparser.ValueType, offset int, err error) {
+		subjects = append(subjects, string(value))
+	}, "subjects")
 	if err != nil {
 		return err
 	}
 
-	return s.pubSub.Subscribe(ctx.Context(), topic, updater)
+	return s.pubSub.Subscribe(ctx.Context(), subjects, updater)
 }
 
 type PublishDataSource struct {
 	pubSub PubSub
 }
 
 func (s *PublishDataSource) Load(ctx context.Context, input []byte, w io.Writer) error {
-	topic, err := jsonparser.GetString(input, "topic")
+	subject, err := jsonparser.GetString(input, "subject")
 	if err != nil {
-		return fmt.Errorf("error getting topic from input: %w", err)
+		return fmt.Errorf("error getting subject from input: %w", err)
 	}
 
 	data, _, _, err := jsonparser.Get(input, "data")
 	if err != nil {
 		return fmt.Errorf("error getting data from input: %w", err)
 	}
 
-	if err := s.pubSub.Publish(ctx, topic, data); err != nil {
+	if err := s.pubSub.Publish(ctx, subject, data); err != nil {
```

Comment:
Have you thought about adding tests that subscribe to the correct topics, depending on the input?